zerver/lib/actions.py

```diff
@@ -2354,11 +2354,16 @@ def fetch_stream_subscriber_user_ids(stream: Stream) -> List[int]:
     for stream in streams:
         new_users = [user for user in users if (user.id, stream.id) in new_streams]
 
-        # Users newly added to invite-only streams need a `create`
-        # notification, since they didn't have the invite-only stream
+        # Users newly added to invite-only streams and acting user if is a relam admin
+        # (in case if acting user is not subscribed to stream)
+        # need a `create` notification, since they didn't have the invite-only stream
         # in their browser yet.
+        event_recievers = [user.id for user in new_users]
+        if acting_user is not None:
+            if acting_user.is_realm_admin:
+                event_recievers.append(acting_user.id)
         if not stream.is_public():
-            send_stream_creation_event(stream, [user.id for user in new_users])
+            send_stream_creation_event(stream, list(set(event_recievers)))
 
```

Comment:
@timabbott Can you review this commit, when you got time? Thanks!

Reply:
yeah, this should be all realm admins

Reply:
Okay, after trying some scenarios locally(creating stream on other tab and login with other user in incognito window), we should only send this event to current realm admin not to all realm admins. Cause if user open overlay of streams, then it do add newly created stream. So it rarely happens where someone create a private stream and on the same time realm admin have streams overlay open in window. 
For the acting-realm-admin user, we have to send otherwise they have to reload or reopen overlay to see created stream.

Reply:
What's the benefit of not sending to all realm admins?

Reply:
We send this event to browser for real time sync, so this is very rare scenario of when any user create a private stream and on the same time another realm admin has open stream settings. Cause only in this case we do require this even, otherwise browser doesn't need this event, on opening stream settings overlay it automatically sync. If we do add, it will generate more events and send to realm admin user's browser client, even if it's not required.

Reply:
The cost of the extra events is very small; less than the cost of sending a new message to a medium-sized stream. So I wouldn't worry about that!

In general, almost all of the updating costs comes from the buddy list, which updates once a minute or so. Small things like updating for events is not an issue.